### ISSUE TITLE
Unit test(s) for stopflow plan

### ISSUE THREAD

**Issue by callumforrester** [DiamondLightSource/i22-bluesky#59]

**Comment by stan-dot** [comment]
@callumforrester could you please elaborate what kind of tests do you have in mind?

**Comment by callumforrester** [comment]
@stan-dot updated issue description

**Comment by callumforrester** [comment]
They specifically only the test the sequence table creating part of the plan, not the poking devices part

**Comment by stan-dot** [comment]
I infer that this means those plans are to be ran on the machine live with real PVs on a testing day, right?

**Comment by callumforrester** [comment]
No, they are _unit_ tests, they test pure functions. The ophyd-async tests I linked only test with mocked devices. The purpose of this ticket is to produce tests that are run in CI and exercise this code regularly to make sure it doesn't break.

**Comment by stan-dot** [comment]
ok, thanks for the clarification. 

would it be good to make predictable testig on the machine day through defining the tests in a similar way though?

this sounds like it would work very similaril

**Comment by stan-dot** [comment]
studying the you provided, @callumforrester 

The way I see it, it tests the interface by putting it in a plan, but the plan is interspersed with `assert` statements.

i22 stopflow is not an interface, but a plan too itself.

here's where from my understanding we can either call the whole plan but then the asserts won't be so interspersed, OR we copy paste the plan and add asserts there but then if the original plan changes it's of little meaning if the previous plan passes.


I think I'd like to gather more examples of how this has been done so far, maybe at MX?

**Comment by stan-dot** [comment]
unassigned as the plan is now towards the linkam plan back to basics so to speak
